search-service/src/main/resources/db/migration/V0_41__migrate_to_creator_right.sql

```diff
@@ -0,0 +1,45 @@
+WITH entities AS (
+    SELECT entity_id, count(*) as admin_right_count
+    FROM entity_access_rights
+    WHERE access_right = 'rCanAdmin'
+    GROUP BY entity_id
+)
+UPDATE entity_access_rights
+SET access_right = 'rIsOwner'
+WHERE entity_id IN (select entity_id from entities where admin_right_count = 1)
+AND access_right = 'rCanAdmin';
+
+-- set isOwner for entities with more than admin right
+WITH entities AS (
+    SELECT entity_id, count(*) as admin_right_count
+    FROM entity_access_rights
+    WHERE access_right = 'rCanAdmin'
+    GROUP BY entity_id
+), entities_more_than_one_admin AS (
+    SELECT entity_id
+    FROM entities
+    WHERE admin_right_count > 1
+), entities_with_oldest_date AS (
+    SELECT entity_id, min(created_at) as created_at
+    FROM temporal_entity_attribute
+    WHERE entity_id IN (select entity_id from entities_more_than_one_admin)
+    GROUP BY entity_id
+), entities_oldest_with_sub AS (
+    select distinct tea.entity_id, sub
+    from temporal_entity_attribute tea, entities_with_oldest_date
+    inner join lateral (
+        select sub
+        from attribute_instance_audit
+        where temporal_entity_attribute = tea.id
+          and time_property = 'CREATED_AT'
+          and sub is not null
+    ) l on true
+    where tea.entity_id = entities_with_oldest_date.entity_id
+    and tea.created_at = entities_with_oldest_date.created_at
+)
```

Comment:
here why did you use `on true` in the inner join ?

Reply:
indeed, it's not needed here since `sub` cannot be null